### PR TITLE
Feature/support config file

### PR DESCRIPTION
**Pull request by tleekkul**
- Added `./bch-devsuite build-config {output_path}`
- Modify `./bch-devsuite init` to read from TOML file instead of showing command line wizard.
- Fixed `./clean` not remove smartbch data dir

### CODE REVIEW

File: docs/configuration-examples.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+## SmartBCH Single Node Private Testnet
+
+```toml
+network = "regtest"
+exposed_ports = true
+docker_network = "bch-devsuite-network"
+
+[local_node]
+node = "bu"
+regtest_blocks = "200"
```
Comment (SuperCipher) [inline docs/configuration-examples.md:68]: Need a comment here that this is for Regtest initial block generation.

File: docs/configuration-examples.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+# Example Chris Troutner JS Tech Stack TOML config file
+
+network = "mainnet"
+rest_service = "bchapi"
+exposed_ports = true
+docker_network = "bch-devsuite-network"
+
+[local_node]
+node = "bu"
+wait_time = 6
```
Comment (SuperCipher) [inline docs/configuration-examples.md:44]: This seems redundant to this https://github.com/ActorForth/bch-devsuite/pull/51/files#diff-cc74d50f1cf8bad20bd110a302a8b4f95245cdca769e1aa8a28e4890f5a4adb9R69

File: docs/configuration-examples.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+```toml
+network = "regtest"
+exposed_ports = true
+docker_network = "bch-devsuite-network"
+
+[local_node]
+node = "bu"
+regtest_blocks = "200"
+wait_time = 6
```
Comment (SuperCipher) [inline docs/configuration-examples.md:69]: Need a comment here that this is a delay before running initializes command.
Need to be high if you run on a slow system.

Feature request:
Or better yet have polling for node readiness

File: bch-devsuite [hunk 4]
```diff
@@ -9,12 +9,14 @@ import subprocess
...
 import time
 from dataclasses import asdict, dataclass
+from datetime import datetime
 from enum import Enum
 from pathlib import Path
 from typing import List, Optional
 from urllib.parse import urlparse
 
 import inquirer
+import toml
```
[the remark this answers is not in the document]
Comment (tleekkul) [reply bch-devsuite:19]: After discussed with Sajal, we think TOML is friendlier with beginner than YAML or JSON.

File: bch-devsuite [hunk 5]
```diff
@@ -28,7 +30,6 @@ BCH_GENERATE_ADDRESS = "bchreg:qzerfrqkvsclmrlx3hzl0j2ac2eh0k88wqa0efygtm"
 BCH_WIF = "cUDjy47QLpAUyv1beivDPoYCHjCRVqwRMwCuSnkdF4teVFBDD1ip"
```
[the remark this answers is not in the document]
Comment (tleekkul) [reply bch-devsuite:30]: This I really don't know, just bring it from old ./setup script

File: bch-devsuite [hunk 6]
```diff
@@ -520,17 +508,54 @@ def finish():
     # TODO: Show open ports for each component
 
 
+def read_config(input_file: Path):
+    global config
+    with open(input_file, "r") as f:
+        cf = toml.loads(f.read())
```
[the remark this answers is not in the document]
Comment (tleekkul) [reply bch-devsuite:514]: Actually it should, but I plan to do it as another enhancement.

File: bch-devsuite [hunk 7]
```diff
@@ -520,17 +508,54 @@ def finish():
...
+        if "smartbch" in cf and cf["smartbch"]:
+            config.smartbch = SmartBCHConf(
+                enabled=cf["smartbch"]["enabled"],
+                test_keys=cf["smartbch"].get("test_keys", []),
+            )
+        config.bch_rpc_conf = RPCConf(
+            host=cf["bch_rpc_conf"]["host"],
+            port=cf["bch_rpc_conf"]["port"],
+            username=cf["bch_rpc_conf"]["username"],
+            password=cf["bch_rpc_conf"]["password"],
```
[the remark this answers is not in the document]
Comment (tleekkul) [reply bch-devsuite:541]: Yeah, and I don't see much benefit for password as env... if we really want to be more secure I think we need much better mechanism than env, I would look at environment variables before anything else if I am the attacker.

File: docs/README.md [hunk 8]
```diff
@@ -25,28 +25,65 @@ source ./venv/bin/activate
...
+
+[smartbch] # SmartBCH
+enabled = true
+test_keys = [ "0b7cdf43329298b26d34d311b25d39f19c60fff25ba45b121284f91e12f17658", "b4d85a7a944b08bab74d0e9e9d612ee409649b382e4de500ee3bd7b7e9c6954f", "216fe772968f326d1b992da744db79fcf06cf6f1142d18086fb4b5a7005cdb8f", "adb378c6b0b9b9cb6190c88cbcaa992388f8e37f1d9c7fc791d08201d04047dc", "09b13dbd311823699802bad7240315021f9e79fe029cc0c0a7a15ab614f303d3", "7c5f4f8eb1f8a82dc9f243350082a1542b2d77d09832023b5cf8f158196a717e", "2917909f71ca82665e6f9ab50b05ecc869f49b9157d0b17976ccd000b3987e29", "989a02864785024b8488d4b22bbdea98048389c18879c18a95e72fbca11c0048", "e473abfa2982915d2cadb204dd41e41afce1b4e8851783a6b1356f5e6784774b", "e3bab3da3a55ac52b241f5d4c2066125b47e197316339540536f66ff92f38585"] # test accounts, only needed for regtest
+
+[bch_rpc_conf]
+host = "http://bch-node" # BCH RPC host, leave it to be "http://bch-node" if you also run BCH local node with bch-devsuite
+port = "8332" # BCH RPC port
+username = "actorforth" # BCH RPC username
+password = "BWrzap0bqMjezoeHtOzgOOcUgzkxaL6w" # BCH RPC password
```
[the remark this answers is not in the document]
Comment (tleekkul) [reply docs/README.md:70]: Ok

File: bch-devsuite (hunk 5 above)
[the remark this answers is not in the document]
Comment (SuperCipher) [reply bch-devsuite:30]: It's doesn't seem to be used anywhere.
So let's remove it.

**Pull request closed without merging**